SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -22,62 +23,109 @@ SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName&
 void SpatialLoadBalanceEnforcer::OnAuthorityIntentComponentUpdated(const Worker_ComponentUpdateOp& Op)
 {
 	check(Op.update.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID);
-	if (StaticComponentView->HasAuthority(Op.entity_id, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
+
+	MaybeQueueAclAssignmentRequest(Op.entity_id);
+}
+
+void SpatialLoadBalanceEnforcer::OnLoadBalancingComponentAdded(const Worker_AddComponentOp& Op)
+{
+	// Should only be passed auth intent or ACL.
+	check(Op.data.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID || Op.data.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+
+	MaybeQueueAclAssignmentRequest(Op.entity_id);
+}
+
+void SpatialLoadBalanceEnforcer::OnLoadBalancingComponentRemoved(const Worker_RemoveComponentOp& Op)
+{
+	// Should only be passed auth intent or ACL.
+	check(Op.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID || Op.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+
+	if (AclAssignmentRequestIsQueued(Op.entity_id))
 	{
-		QueueAclAssignmentRequest(Op.entity_id);
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Log,
+			TEXT("Component %d for entity %lld removed. Can no longer enforce the previous request for this entity."),
+			Op.component_id, Op.entity_id);
+		AclWriteAuthAssignmentRequests.Remove(Op.entity_id);
```

Comment:
Since AclAssignmentRequestIsQueued is hiding AclWriteAuthAssignmentRequests here, I think we should also have a RemoveQueuedAclAssignmentRequest as well.   Similarly below on line 58 and 74.

Alternatively, make a RemoveAssingmentRequestIfQueued which takes a log message and wrap the whole thing.

Reply:
will leave this for now- raised this ticket: https://improbableio.atlassian.net/browse/UNR-3020 to remove the queue and simplify the load balance enforcer's logic.